src/components/Dashboard/PendingList/PendingTemplate.vue

```diff
@@ -57,23 +60,43 @@
 </template>
 
 <script lang="ts">
-import { Component, Prop, Vue } from 'vue-property-decorator'
+import { Component, Prop, Mixins } from 'vue-property-decorator'
 import { DateTooltip } from '@/components/common'
+import { DateMixin } from '@/mixins'
+import { Getter } from 'pinia-class'
+import { useRootStore } from '@/stores'
 
 @Component({
   components: { DateTooltip }
 })
-export default class PendingTemplate extends Vue {
+export default class PendingTemplate extends Mixins(DateMixin) {
   @Prop({ required: true }) readonly item!: any
   @Prop({ required: true }) readonly index!: number
 
+  @Getter(useRootStore) getNameRequest!: any
+
   get submitter (): Date {
     return this.item.header.submitter
   }
 
   get submittedDate (): Date {
     return new Date(this.item.header.date)
   }
+
+  expiresText (nameRequest: any): string {
+    const date = this.apiToDate(nameRequest.expirationDate)
+    const expireDays = this.daysFromToday(date)
+    // NB: 0 means NR expires today
+    if (isNaN(expireDays) || expireDays < 0) {
+      return 'Expired'
+    } else if (expireDays < 1) {
+      return 'Expires today'
+    } else if (expireDays < 2) {
+      return 'Expires tomorrow'
+    } else {
+      return `Expires in ${expireDays} days`
+    }
+  }
```

Comment:
This is a duplicate of a method in TodoList.vue. I recommend that you move it to a mixin (date mixin?) and change its callers accordingly.

Reply:
Moved it to date Mixin and updated